### PR TITLE
CB-95: Use SimpleMDE as an editor for CritiqueBrainz

### PR DESCRIPTION
**Pull request by eshansingh**
[JIRA ticket](https://tickets.metabrainz.org/browse/CB-95).

This modifies `templates/review/modify/base.html` to use `SimpleMDE` for the review content editor.

Also, there are not nearly as many actual changes as the diff indicates - the changes to `npm-shrinkwrap.json` account for most of them.

### CODE REVIEW

**Comment by eshansingh** [comment]
Just realized an issue with the fullscreen mode, not sure how to deal with it right now:

fullscreen SimpleMDE issue

**Comment by eshansingh** [comment]
Set the z-index of the editor toolbar when fullscreen to `1000`. Bit hacky, but solves the issue.

File: critiquebrainz/frontend/static/scripts/wysiwyg-editor.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+var SimpleMDE = require('simplemde');
+
```
Comment (ferbncode) [inline critiquebrainz/frontend/static/scripts/wysiwyg-editor.js]: I think there is no need of that newline.

File: critiquebrainz/frontend/templates/review/modify/base.html [hunk 2]
```diff
@@ -51,68 +32,51 @@
         <div class="col-sm-12">
           {% for choice in form.license_choice %}
             <div class="radio">
-              <label>{{ choice }}{{ choice.label.text | safe }}</label>
+                <label>{{ choice }}{{ choice.label.text | safe }}</label>
```
Comment (ferbncode) [inline critiquebrainz/frontend/templates/review/modify/base.html]: You need to fix the indentation changes here. Make sure you use 2 spaces for `.html` files for indentation. These changes feel a bit like text-editor changes.

**Comment by ferbncode** [comment]
@brainzbot whiltelist

**Comment by eshansingh** [comment]
@ferbncode Noted - working on changes.

**Comment by mwiencek** [comment]
@brainzbot, add to whitelist

**Comment by mwiencek** [comment]
@brainzbot, add to whitelist

**Comment by eshansingh** [comment]
@ferbncode I'm also wondering whether I might as well integrate [CB-239](https://tickets.metabrainz.org/browse/CB-239) into this PR because it's such a small and closely related change.

**Comment by ferbncode** [comment]
@eshansingh I would suggest to make a different PR as it is listed as a separate issue on JIRA.

**Comment by eshansingh** [comment]
@ferbncode Alright, I'll do that separately.

**Comment by anshuman73** [comment]
yeah, @eshansingh I'd suggest doing that as soon as this PR is merged, allowing you to not re-base this one.
Another PR also helps maintain the history, in case we ever need to track it

**Comment by eshansingh** [comment]
@anshuman73 So, are any further changes needed for this PR?

File: critiquebrainz/frontend/templates/review/modify/base.html [hunk 3]
```diff
@@ -57,61 +44,44 @@
           <br /><small class="text-warning"><em>{{ _('You cannot change the license after the review is published.') }}</em></small>
         </div>
       </div>
-    {% endif %}
-    {% block license_agreement_input %}{% endblock %}
+     {% endif %}
+     {% block license_agreement_input %}{% endblock %}
```
Comment (anshuman73) [inline critiquebrainz/frontend/templates/review/modify/base.html]: @eshansingh it seems some indentation is still left here, can you fix it ?

File: critiquebrainz/frontend/templates/review/modify/base.html [hunk 4]
```diff
@@ -57,61 +44,44 @@
...
-        if (beingSaved || (oldData == $('#review-editor').serialize())) {
-          return undefined;
-        }
-        var confirmationMessage = "Your review has not been saved.";
-        (e || window.event).returnValue = confirmationMessage;     // Gecko and Trident
-        return confirmationMessage;                                // Gecko and WebKit
+          if (beingSaved || (oldData == $('#review-editor').serialize())) {
+              return undefined;
+          }
+          var confirmationMessage = "Your review has not been saved.";
```
Comment (anshuman73) [inline critiquebrainz/frontend/templates/review/modify/base.html]: Out here as well.

File: critiquebrainz/frontend/templates/review/modify/base.html [hunk 5]
```diff
@@ -57,61 +44,44 @@
...
-              });
-            }
-          } else {
-            $("#preview").html('<em class="text-muted">' + {{ _('Loading preview...') | tojson }} + '</em>');
-          }
-        });
       });
+      if ('{{not review or review.is_draft}}'){
+          $("#btn-draft").click(function(){
+              beingSaved = true;
```
Comment (anshuman73) [inline critiquebrainz/frontend/templates/review/modify/base.html]: Maybe use 2 spaces in indentation in JS scripts everywhere

Comment (anshuman73) [reply critiquebrainz/frontend/templates/review/modify/base.html]: Just like for HTML

File: critiquebrainz/frontend/templates/review/modify/base.html [hunk 6]
```diff
@@ -86,32 +77,11 @@
       });
       if ('{{not review or review.is_draft}}'){
         $("#btn-draft").click(function(){
-          beingSaved = true;
-          $("#state").attr('value', 'draft');
-          $("#review-editor").submit();
+            beingSaved = true;
```
Comment (anshuman73) [inline critiquebrainz/frontend/templates/review/modify/base.html]: Still some left :(

Comment (eshansingh) [reply critiquebrainz/frontend/templates/review/modify/base.html]: Wait, I don't see it - it's 2 spaces, right?

Comment (eshansingh) [reply critiquebrainz/frontend/templates/review/modify/base.html]: Nope, it's wrong :( I promise this kind of stuff won't happen in the future. My editor went mad on these.

Comment (anshuman73) [reply critiquebrainz/frontend/templates/review/modify/base.html]: Haha, no issues

**Comment by anshuman73** [comment]
Considering The ajax call to the preview endpoint is no longer needed as the preview can now be rendered in the editor itself, I think it would be safe to remove this - 

https://github.com/metabrainz/critiquebrainz/blob/master/critiquebrainz/frontend/views/review.py#L259

**Comment by anshuman73** [comment]
Also, thanks for squashing the indentation commits :)

**Comment by eshansingh** [comment]
@anshuman73

**Comment by anshuman73** [comment]
@eshansingh haha, that's okay too. same thing (almost) - rewriting the history.

The PR is good to go from my side :)

**Comment by eshansingh** [comment]
🎉 :shipit: @ferbncode

**Comment by eshansingh** [comment]
Sorry, spotted a typo in `aefe9db` (SimpleMD `->` SimpleMDE) and used `reword`.

**Comment by paramsingh** [comment]
@ferbncode do you want me to merge this? There's lots of open PRs in CB that I'd like to help close if you want?

**Comment by anshuman73** [comment]
@paramsingh that'll be great. @ferbncode and I have both approved the PR and think it's ready to go, though we'd love you to do a quick review too, if you'd liek (<--hehe) and then merge it :)

**Comment by anshuman73** [comment]
P.S. Squashing them wile merging might be a good idea too, considering there's a lot of commit messages.

**Comment by paramsingh** [comment]
:shipit:

**Pull request merged**